python/python/res/util/res_log.py

```diff
@@ -1,19 +1,26 @@
 from res.util import ResUtilPrototype
+from res.util.enums import MessageLevelEnum
 
 
 class ResLog(object):
-    _init = ResUtilPrototype("void res_log_init_log(int, char*, bool)", bind=False)
-    _write_log = ResUtilPrototype("void res_log_add_message_py(int, char*)", bind=False)
+    _init = ResUtilPrototype("void res_log_init_log(message_level_enum, char*, bool)", bind=False)
+    _write_log = ResUtilPrototype("void res_log_add_message_str(message_level_enum, char*)", bind=False)
     _get_filename = ResUtilPrototype("char* res_log_get_filename()", bind=False)
 
+    @staticmethod
+    def __enum(log_level):
+        if isinstance(log_level, int):
+            log_level = MessageLevelEnum.to_enum(log_level)
```

Comment:
What to do if the `isinstance(log_level, int)` test fails - exception?

Reply:
This was actually just under the assumption that the type is either int or a MessageLevelEnum.

Reply:
I guess that is why it is slightly dangerous to start with `isinstance( )` checks in Python, it kind of breaks the duck-typing contract. Maybe the `MessageLevelEnum.to_enum( )` could be modified to accept a enum instance as input argument, then that method can be called unconditionally?